### PR TITLE
Make req.auth.require discardable

### LINKED ISSUE #2428: Request.Authentication.require discardable
Would be good good if `auth.require` was marked with `@discardableResult` so you can write

```swift
try req.auth.require(SomeAuthenticatable.self) 
```

instead of 

```swift
let _ = try req.auth.require(SomeAuthenticatable.self) 
```

when you're only interested in that some authenticated user is present and not the exact user

### PR DESCRIPTION
**Pull request by SpencerCurtis**
Adds `@discardableResult` attribute to `req.auth.require` to allow ignoring the result without a warning (#2446, fixes #2428). 

```swift
try req.auth.require(User.self) 
```

### CODE REVIEW

**Comment by SpencerCurtis** [comment]
Totally, I saw the comment in (#2428). I just figured since it was added to the To Do board I figured I'd quickly implement it.

**Comment by MrLotU** [comment]
Hmm, not sure why it was added to the board since IMO it shouldn't be implemented. CC @tanner0101

File: Sources/Vapor/Authentication/AuthenticationCache.swift [hunk 1]
```diff
@@ -32,7 +32,7 @@ extension Request.Authentication {
     /// Returns an instance of the supplied type. Throws if no
     /// instance of that type has been authenticated or if there
     /// was a problem.
-    public func require<A>(_ type: A.Type = A.self) throws -> A
+    @discardableResult public func require<A>(_ type: A.Type = A.self) throws -> A
```
Comment (tanner0101) [inline Sources/Vapor/Authentication/AuthenticationCache.swift:35]: newline after `@discardableResult`

**Comment by tanner0101** [comment]
After the nit, this is good to merge, thanks! Please update the PR for release: https://github.com/vapor/vapor/blob/master/.github/contributing.md#releases

**Comment by SpencerCurtis** [comment]
@tanner0101 I've updated the title and description. I'm happy to re-edit if it isn't quite in-line with the way it should be.

**Comment by 0xTim** [comment]
FYI this is a semver-patch not minor so I've updated the label

**Comment by tanner0101** [comment]
These changes are now available in [4.26.2](https://github.com/vapor/vapor/releases/tag/4.26.2)

**Pull request merged**